### ISSUE TITLE
findResource("*.theme") always queries the server

### ISSUE THREAD

**Issue by childsb** [maqetta/maqetta#886]
tough fix here, but the findResource(*.theme) will always hit the server.  Search should be conducted client side, and only load elements that haven't been loaded from server.

i'm guessing its like this because there are file wildcard libraries already in Java, so the logic behind the matches didn't have to be implemented.  Doing this server side will require parsing the \* and ? wildcards, and figuring out the string/path logic in JavaScript.

**Comment by JonFerraiolo** [comment]
@billreed63

**Comment by JonFerraiolo** [comment]
-> @peller M7 Prio-Med - might be a dupe of other bugs

**Comment by peller** [comment]
Hopefully, we can avoid making this call often.  May not be necessary to make this optimization

-> M8

**Comment by billreed63** [comment]
I have changed the library.getThemes code so that the server is only hit when a theme is added/deleted and on the first getThemes on the client. The themes are now cached and the caches is only invalidated on theme add/delete/rename

**Comment by billreed63** [comment]
Added a getThemes command on the server that returns the list of themes in the user workspace as well as the theme data from the theme files as one json object. 

The File resource for the theme file is loaded on demand the first time it is accessed by the code

**Comment by peller** [comment]
Regarding this comment:

```
      var targetFile = systemResource.findResource(cssFile.url).getPath(); // target //FIXME: can we skip findResource step?
```

Is the findResource call just to accomplish some sort of path math?  Is there a way to determine this without going through findResource?

**Comment by billreed63** [comment]
It is for path math, I know of no other way to find the location of the theme files, findResource is cached to once we find the resource the first time it is cached on the client and all future calls for that resource are not calls to the server

**Comment by peller** [comment]
Perhaps if we could just declare a place where theme files live (even a path relative to the user's root) we could ship that down with the initialization info and avoid this findResource call.

**Comment by billreed63** [comment]
fixed by f2778dae0ae776425f096b231cc8ff6f7fca39ce
